### PR TITLE
JSS Styles rework

### PR DESCRIPTION
**Pull request by nfour**
Intention is to further componentize the project, getting away from CSS dependencies.

- [x] Demonstrate JSS 
- [x] Convert components to JSS
  - [x] BodyContent
  - [x] BodySchema
  - [x] BodySchemaSwitcher
  - [x] ContentContainer
  - [x] CopyButton
  - [x] Description
  - [x] Example
  - [x] ExternalLink
  - [x] Header
  - [x] Indicator
  - [x] Method
  - [x] Navigation
  - [x] NavigationMethod
  - [x] NavigationTag
  - [x] Overlay
  - [x] Page
  - [x] Property
  - [x] PropertyConstraints
  - [x] Response
  - [x] SecurityContainer
  - [x] ServiceContainer
  - [x] Base
  - [x] Demo
  - [x] general/base scss
  - [x] Refactor all variables, themes

## Features
- [x] Modular CSS
- [x] Static theming support
- [x] ~~Dynamic theming support~~
    - Cant support, pending upstream PR's

## Deps
- [x] Removed `sass-loader` `node-sass` `balloon-css`

### CODE REVIEW

**Comment by nfour** [comment]
Examples:

```js
// Foo.styles.js
import { createSheet } from '../../theme'

export const styles = createSheet((theme) => ({
  foo: {
    background: 'transparent',
    color: 'white'
  }
}))
```

```js
// Foo.js
import { styles } from './Foo.styles'

@styles
export default class Foo extends Component {
  render () {
    const { classes } = this.props

    return <div className={classes.foo}></div>
  }
}
```

**Comment by brendo** [comment]
Looking good man, checked it out locally to have a play around 👍 

Did you want a hand with the remaining components or happy just knocking them off one by one?

**Comment by brendo** [comment]
Note JSS and Jest snapshots might need some work.

https://github.com/styled-components/jest-styled-components/issues/29
https://github.com/cssinjs/react-jss/issues/93

**Comment by nfour** [comment]
The remainder of this MR is busywork, feels like 20% left to do in terms of time put in. 

The testing stuff IMO just needs to pass for the components, thus if classNames can be mocked, we will be fine. Snapshotting JSS can come in after those PR's are in, in jest-styled-components.

File: jest.config.js [hunk 1]
```diff
@@ -7,8 +7,9 @@ module.exports = {
     'node_modules'
   ],
   moduleNameMapper: {
-    'balloon-css': '<rootDir>/src/__mocks__/styleMock.js',
-    '\\.(css|scss)$': '<rootDir>/src/__mocks__/styleMock.js',
+    'balloon-css': 'identity-obj-proxy',
```
Comment (brendo) [inline jest.config.js]: remove

**Review by brendo — APPROVED** [review]

**Pull request merged**